test_data/std/permutation_via_challenges.asm

```diff
@@ -12,8 +14,16 @@ machine Main with degree: 8 {
     col witness b1(i) query Query::Hint(fe(7 - i));
     col witness b2(i) query Query::Hint(fe(7 - i + 42));
 
+    let permutation_constraint = Constr::Permutation(
+        Option::Some(first_four),
```

Comment:
@chriseth Is there a reason this is an option? You can always pass `1` as the expression.

Reply:
This originated in the parser (`a {b, c}` vs `{b, c}`) but it could also be an efficiency thing, it saves you a multiplication. But yeah, not too strong on this one.

Reply:
Hmm, if you wanted to optimize performance, you could check for `selector == 1`, right? Or do we not have equality for `expr`? Using `Option::None` and `Option::Some(1)` as the selector is the same thing, so I don't thing there should be two ways to specify it.

Is the `Constr::Permutation` enum used somewhere? Or can I change it in this PR? Then, I don't need to handle the `selector == None` case.

Reply:
BTW, when multiplying by 1 in an identity, I think the optimizer is smart enough to just remove it. I wonder if we could just do the same in the hint expressions somehow, so we save computation when computing the hint.

Reply:
The optimizer currently only operates on identities. The main reason is that in the identities all function calls are inlined, so the optimizer can be implemented with a greedy algorithm. For generic code, we could also implement some optimizations, but the more generic our code becomes (e.g. implementing Add traits etc), we have to think about inlining.

Reply:
I think the enum is used in the condenser, so feel free to change it. I was also thinking about adding a SelectedExpressions struct - but we don't have structs yet, so it would be a single-variant enum which is not nice.

Reply:
Leaving at it is for now, at least in this PR :)